src/Grpc/Internal/GrpcAsyncCallSpecimenBuilder.cs

```diff
@@ -0,0 +1,122 @@
+﻿using System;
+using System.Threading.Tasks;
+using AutoFixture.Kernel;
+using Grpc.Core;
+using Grpc.Core.Testing;
+
+namespace Kralizek.AutoFixture.Extensions.Internal
+{
+    public class GrpcAsyncCallSpecimenBuilder : ISpecimenBuilder
+    {
+        private readonly CallParameters _callParameters;
+        
+        public GrpcAsyncCallSpecimenBuilder(CallParameters callParameters)
+        {
+            _callParameters = callParameters ?? throw new ArgumentNullException(nameof(callParameters));
+        }
+
+        public object Create(object request, ISpecimenContext context)
+        {
+            if (context is null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            return request switch
+            {
+                null => throw new ArgumentNullException(nameof(request)),
+                Type { IsGenericType: true } type when type.GetGenericTypeDefinition() == typeof(AsyncServerStreamingCall<>) => CreateAsyncServerStreamingCall(type, context, _callParameters),
+                Type { IsGenericType: true } type when type.GetGenericTypeDefinition() == typeof(AsyncUnaryCall<>) => CreateAsyncUnaryCall(type, context, _callParameters),
+                Type { IsGenericType: true } type when type.GetGenericTypeDefinition() == typeof(AsyncClientStreamingCall<,>) => CreateAsyncClientStreamingCall(type, context, _callParameters),
+                Type { IsGenericType: true } type when type.GetGenericTypeDefinition() == typeof(AsyncDuplexStreamingCall<,>) => CreateAsyncDuplexStreamingCall(type, context, _callParameters),
+                _ => new NoSpecimen()
+            };
+        }
+
+        private static object CreateAsyncServerStreamingCall(Type type, ISpecimenContext context, CallParameters callParameters)
+        {
+            var typeArgument = type.GenericTypeArguments[0];
+
+            var asyncStreamReaderType = typeof(IAsyncStreamReader<>).MakeGenericType(typeArgument);
+
+            var responseStream = context.Resolve(asyncStreamReaderType);
```

Comment:
Fair point. What would be the best way to handle it? I'm thinking of some kind of informative exception but not sure what's the best approach.

Reply:
Do you think something like this would be enough?

```csharp
var responseStream = context.Resolve(asyncStreamReaderType) switch
{
    var x when x.GetType().GetGenericTypeDefinition() == typeof(IAsyncStreamReader<>) => x,
    _ => throw new Exception($"Unable to resolve {asyncStreamReaderType}")
};
```

Reply:
Should be enough, but I'm not sure you should throw though. If it's an `OmitSpecimen` you should probably forward that. It means the user deliberately chose to omit instances of this type. Probably same for `NoSpecimen`.

BTW have you considered what would happen if the `IAsyncStreamReader<>`, for example, is frozen?

You could easily implement your own internal `TryResolve` extension method, if you think it might help you.

Reply:
But if the user chose to omit those types, it's impossible to proceed 🤔 

Anywyay, I foresee users freezing the enumerable of items rather than the stream reader because freezing the reader doesn't give you access to the raw items. 

See: https://github.com/Kralizek/AutoFixtureExtensions/pull/3/files#diff-f7c01fb34b74ad5fc0219e856b87397701ea07a322ad9fda1b59bbee2866f7fbR46-R55

Ideally, I would have hoped for AutoMoq autowiring to kick in and skipping explicitly setting up the response but it didn't work.

Reply:
@Kralizek of course if it doesn't make sense to continue, it is better to throw.